### ISSUE TITLE
Thanks for the great project!

### ISSUE THREAD

**Issue by manwithsteelnerves** [Mamantenok1599/UnityVATBaker#1]
I'm currently working on the same but used StruturedBuffers instead of baking in textures.

Have you tried using buffers and accessing in the shader directly instead of going through baking in the textures?
Do you see any difference?

Also, how better it performs compared to GPU Skinning?

**Comment by Mamantenok1599** [comment]
Hello! How exactly do you store data from a Structured Buffer on disk for further processing?
As far as I understand, the SkinnedMeshRenderer already uses GPU Skinning. However, I haven't studied this topic in depth, so I can't say for sure.

**Comment by manwithsteelnerves** [comment]
> Hello! How exactly do you store data from a Structured Buffer on disk for further processing? As far as I understand, the SkinnedMeshRenderer already uses GPU Skinning. However, I haven't studied this topic in depth, so I can't say for sure.

Well, I stored the mesh data(as a list of vertices, indices) and submitted it via MaterialPropertyBlock to shader and rendered with RenderPrimitives. This is actually in a way similar to the texture approach you have but don't have limitations about packing and precision.

However, wondering if Structured Buffer memory access is better or Texture reading or Skinning directly on the GPU by passing bone matrices. If I need to know the perf difference, I need to make all three and compare which is very tedious. So checking if you have any details about it :)

**Comment by Mamantenok1599** [comment]
> > Hello! How exactly do you store data from a Structured Buffer on disk for further processing? As far as I understand, the SkinnedMeshRenderer already uses GPU Skinning. However, I haven't studied this topic in depth, so I can't say for sure.
> 
> Well, I stored the mesh data(as a list of vertices, indices) and submitted it via MaterialPropertyBlock to shader and rendered with RenderPrimitives. This is actually in a way similar to the texture approach you have but don't have limitations about packing and precision.
> 
> However, wondering if Structured Buffer memory access is better or Texture reading or Skinning directly on the GPU by passing bone matrices. If I need to know the perf difference, I need to make all three and compare which is very tedious. So checking if you have any details about it :)

That is, do you "bake" the data into a structured buffer at the start of the game and then sample the data via MPB? Or do you store this data as .asset?

Which of these is faster, I can't tell you :) I need to test it too)

**Comment by manwithsteelnerves** [comment]
> That is, do you "bake" the data into a structured buffer at the start of the game and then sample the data via MPB? Or do you store this data as .asset?
Yes. I do store as .asset files and initialize the structured buffers runtime.

> Which of these is faster, I can't tell you :) I need to test it too)
Yea, I understand :)
Hopefully, will find an answer soon :)
